code/modules/atmospherics/machinery/components/fusion/hfr_defines.dm

```diff
@@ -55,10 +43,126 @@
+//  - A maximum power current damper should be enough to always turn a reaction endothermic, even at 100% iron
+//
+// The current configuration has the following properties:
+//  - Cycles from 0 to 5.5/6.15, with the first cycle width over 10000 gas power, then 4000, then 3250, then 2500.
+//    This is enough to mean the initial power up should be very stable, with thresholds at usable points following
+//  - 100% iron drops -2, 1000kW Current damper adds +10:
+//    - 400kW will destabilize a no-iron system half of the time
+//    - 800kW will always destabilize a no-iron system (at the cost of using quite a lot of power while active)
+//    - 1000kW will always destabilize any system (at the cost of using an aburd amount of power while active)
+
+/// Maximum instability before the reaction goes endothermic
+#define FUSION_INSTABILITY_ENDOTHERMALITY 8
+/// How much instability does each kW of current damper add?
+#define HYPERTORUS_INSTABILITY_DAMPER_FACTOR 0.01
+/// How much instability does 100% iron subtract?
+#define HYPERTORUS_INSTABILITY_IRON_FACTOR 2
+
+/// Used to reduce the gas_power to a more useful amount
+#define INSTABILITY_GAS_POWER_FACTOR 0.00025
+/// Used to calculate the toroidal_size for the instability
+#define TOROID_VOLUME_BREAKEVEN 1000
+
+//
+// Damage source: Too much mass in the fusion mix at high fusion levels
+//
+
+// Currently, this is 2700 moles at 1 Kelvin, linearly scaling down to a maximum of 1800 safe moles at 1e8 degrees kelvin
+// Settings:
+/// Start taking overfull damage at this power level
+#define HYPERTORUS_OVERFULL_MIN_POWER_LEVEL 6
+/// Take 0 damage beneath this much fusion mass at 1 degree Kelvin
+#define HYPERTORUS_OVERFULL_MAX_SAFE_COLD_FUSION_MOLES 2700
+/// Take 0 damage beneath this much fusion mass at FUSION_TEMPERATURE_MAX degrees Kelvin
+#define HYPERTORUS_OVERFULL_MAX_SAFE_HOT_FUSION_MOLES 1800
+// From there, how quickly should things get bad?
+/// Every 200 moles, 1 point of damage per second
+#define HYPERTORUS_OVERFULL_MOLAR_SLOPE (1/200)
+// Derived:
+// Given these settings, derive the rest of the equation.
+// Damage is the dependent variable, fusion_moles and damage_source_temperature are the independent variables
+// So the equation takes the form:
+//   damage = molar_slope * fusion_moles + temperature_slope * damage_source_temperature + constant
+// Derive these constants here for readability
+// Derive the temperature slope from the molar slope
+#define HYPERTORUS_OVERFULL_TEMPERATURE_SLOPE (HYPERTORUS_OVERFULL_MOLAR_SLOPE * (HYPERTORUS_OVERFULL_MAX_SAFE_COLD_FUSION_MOLES - HYPERTORUS_OVERFULL_MAX_SAFE_HOT_FUSION_MOLES) / (FUSION_MAXIMUM_TEMPERATURE - 1))
+// Derive the constant to set damage = 0 at our desired thresholds above
+#define HYPERTORUS_OVERFULL_CONSTANT (-(HYPERTORUS_OVERFULL_MOLAR_SLOPE * HYPERTORUS_OVERFULL_MAX_SAFE_HOT_FUSION_MOLES + HYPERTORUS_OVERFULL_TEMPERATURE_SLOPE * FUSION_MAXIMUM_TEMPERATURE))
```

Comment:
This is somewhat insane

Reply:
Kinda? I'm not immediately sure what could be done to improve it, as more intermediate factors are probably going to make it worse, and I really like having human readable constants to set the properties for these kinds of functions. I'm pretty sure the old damage nerf that effectively killed the active range was an accident, and would like to avoid that in the future